app/resources/views.py

```diff
@@ -12,51 +13,105 @@ def index():
-                              city=form.locality.data,
-                              state=form.administrative_area_level_1.data,
-                              zip_code_id=zip_code.id)
-            db.session.add(address)
-            db.session.commit()
-        # Based on form and address id, create a new resource.
-        # TODO: Create helper method in models/resource.py.
-        resource = Resource(name=form.name.data,
-                            description=form.description.data,
-                            website=form.website.data,
-                            address_id=address.id,
-                            user_id=int(current_user.get_id()))
-        db.session.add(resource)
-        db.session.commit()
-        return redirect(url_for('resources.index'))
-    return render_template('resources/add_resource.html', form=form)
+        address = Address.create_address(name,
+                                         street_address,
+                                         city,
+                                         state)
+        address.zip_code = zip_code
+        description = form.description.data
+        website = form.website.data
+        resource = Resource.create_resource(name,
+                                            description,
+                                            website)
+        resource.address = address
+        resource.user = User.query.get(current_user.id)
+        return redirect(url_for('resources.read_resource',
+                                resource_id=resource.id))
+    return render_template('resources/create_resource.html', form=form)
+
+
+@resources.route('/read/<int:resource_id>')
+@login_required
+def read_resource(resource_id):
+    resource = Resource.query.get_or_404(resource_id)
+    # TODO: check on reviews, which is dynamic?
+    return render_template('resources/read_resource.html',
+                           resource=resource,
+                           reviews=resource.reviews,
+                           current_user_id=current_user.id)
 
 
-@resources.route('/resource/<int:resource_id>')
+@resources.route('/review/create/<int:resource_id>', methods=['GET', 'POST'])
 @login_required
-def show_resource(resource_id):
-    """
-    Show the resource with the given id, the id is an integer.
-    """
+def create_review(resource_id):
     resource = Resource.query.get_or_404(resource_id)
-    address = Address.query.get(resource.address_id)
-    user = User.query.get(resource.user_id)
-    return render_template('resources/view_resource.html', resource=resource,
-                           address=address, user=user)
+    form = ReviewForm()
+    if form.validate_on_submit():
+        review = ResourceReview(timestamp=datetime.now(),
+                                content=form.content.data,
+                                rating=form.rating.data)
+        review.resource = resource
+        review.user = current_user
```

Comment:
let's remove the jank and go with resource.user_id = current_user.id or resource.user = current_user._get_current_object(), your choice sir